Description: Use debhelper CPPFLAGS CFLAGS LDFLAGS in build
Author: Mateusz Łukasik <mati75@linuxmint.pl>

--- a/Makefile
+++ b/Makefile
@@ -1,14 +1,14 @@
-CC?=cc
-CFLAGS?=-O
+CC?+=cc
+CFLAGS?+=-O
 LIBS=`[ \`uname\` = "SunOS" ] && echo -lsocket -lnsl -lsendfile`
 
 all: darkhttpd
 
 darkhttpd: darkhttpd.c
-	$(CC) $(CFLAGS) $(LDFLAGS) $(LIBS) darkhttpd.c -o $@
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $(LIBS) darkhttpd.c -o $@
 
 darkhttpd-static: darkhttpd.c
-	$(CC) -static $(CFLAGS) $(LDFLAGS) $(LIBS) darkhttpd.c -o $@
+	$(CC) -static $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $(LIBS) darkhttpd.c -o $@
 
 clean:
 	rm -f darkhttpd core darkhttpd.core darkhttpd-static darkhttpd-static.core
