From e292109d4f4fd32c43d19dae073c98351fc46300 Mon Sep 17 00:00:00 2001
From: Cameron Norman <CameronNemo@users.noreply.github.com>
Date: Wed, 11 Feb 2015 19:27:53 -0800
Subject: [PATCH] Create pidfile before dropping priveleges

---
 darkhttpd.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/darkhttpd.c b/darkhttpd.c
index ea20e9a..d382255 100644
--- a/darkhttpd.c
+++ b/darkhttpd.c
@@ -2538,6 +2538,10 @@ int main(int argc, char **argv) {
         printf("chrooted to `%s'\n", wwwroot);
         wwwroot[0] = '\0'; /* empty string */
     }
+
+    /* create pidfile */
+    if (pidfile_name) pidfile_create();
+
     if (drop_gid != INVALID_GID) {
         gid_t list[1];
         list[0] = drop_gid;
@@ -2553,9 +2557,6 @@ int main(int argc, char **argv) {
         printf("set uid to %d\n", (int)drop_uid);
     }
 
-    /* create pidfile */
-    if (pidfile_name) pidfile_create();
-
     if (want_daemon) daemonize_finish();
 
     /* main loop */
